curl/src/client.cc

```diff
@@ -207,34 +214,26 @@ namespace leatherman { namespace curl {
             // Perform the request
             result = curl_easy_perform(_handle);
             if (result != CURLE_OK) {
-                cleanup_temp_file(temp_path, fp, req, file_path, std::string(errbuf));
-                throw http_file_download_exception(req, file_path, errbuf);
+                throw http_request_exception(req, errbuf);
             }
             fclose(fp);
 
-            LOG_DEBUG("download completed, now writing result to file");
+            LOG_DEBUG("download completed, now writing result to file {1}", file_path);
             fs::rename(temp_path, file_path);
         } catch (http_file_download_exception& e) {
             // so that the code in the lower catch block is not hit
             throw e;
         } catch (http_request_exception& e) {
-            cleanup_temp_file(temp_path, fp, req, file_path, e.what());
+            fclose(fp);
```

Comment:
I see, so http_request_exceptions imply that you need to remove the temp file, while raw file_download_exception means that the temp file is gone or not relevant. Yeah, looks good to me @MikaelSmith